ADR/0047-gitops-onboarding-redesign.md

```diff
@@ -0,0 +1,279 @@
+5. Developer is prompted by UI to install / open repo in VS Code
+3. Launches Konflux plugin and uses it to configure new components, tests, releasePlans
+6. Plugin generates YAML objects in repo
+7. Developer commits and pushes code (optionally via PR, with CI,reviews etc.)
+9. ArgoCD automatically syncs merged changes to the namespace
+
+### Multi-Repo GitOps Model & Optional Services
+
+**GitOps Registration Service (Optional):**
+Konflux will optionally provide a **GitOps Registration Service** API that automates the registration and lifecycle of GitOps repositories as first-class tenants. This service is recommended for production deployments but not required for basic Konflux functionality.
+
+**ArgoCD Service (Optional):**
+ArgoCD can be optionally deployed as a continuous GitOps synchronization service. While recommended for production use, Konflux can operate without ArgoCD, though this results in reduced automation and requires manual synchronization of GitOps repositories.
+
+**Impact of Optional Services:**
+- **With both services**: Full automated GitOps experience with continuous deployment and streamlined onboarding
+- **Without both services**: Konflux remains fully functional but requires manual GitOps repository setup and synchronization. This configuration provides:
+  - Smaller deployment footprint suitable for lower-spec machines
+  - Simplified setup for testing and development environments  
+  - Reduced operational complexity for teams preferring manual control
+  - Degraded user experience with manual setup steps
+  - Reduced disaster recovery capabilities due to lack of automated drift detection
+
+**Tenant Model:**
+- **Tenant Definition**: A tenant is equivalent to a Kubernetes namespace
+- **1:1 Mapping**: Each GitOps repository maps to exactly one Kubernetes namespace
+- **Namespace Declaration**: The namespace name is a parameter to the registration service. Once the repo is registered, the namespace it is immutable.
+- **Registration Immutability**: Re-registering the same repo is rejected to prevent namespace conflicts
+
+**Resource Scope:**
+- **Multi-Resource Support**: One GitOps repo can contain multiple applications, components, integrationTestScenarios, and releasePlans
+- **Single Namespace Constraint**: All resources from a GitOps repo deploy to its designated namespace only
+- **No Cross-Namespace Interaction**: Changes affecting N namespaces require changes in N separate GitOps repos
+
+**Registration Process (when using optional services):**
+1. Team registers GitOps repo via VS Code plugin, CLI (`konflux register`), or UI
+2. at registration time the desired namespace name is provided to (UI | API)
+3. GitOps Registration Service validates the repo and the requested namespace
+4. Service provisions the new requested Kubernetes namespace
+6. Service configures ArgoCD to monitor the repo for continuous deployment
```

Comment:
I'm not sure an additional service is needed to configure argo. Why not creating argo resources directly?

Reply:
Who's going to create it?

Reply:
The UI can also create the ArgoCD applications directly through the k8s api.

Reply:
yeah, you aren't the only one to bring it up. I am updating the relationship for gitops repo to namespace to be 1:many